tests/conftest.py

```diff
@@ -1357,3 +1357,38 @@ def get_reboot_cause(duthost):
     uptime_end = duthost.get_up_time()
     if not uptime_end == uptime_start:
         duthost.show_and_parse("show reboot-cause history")
+
+def collect_db_dump_on_duts(request, duthosts):
+    '''
+        When test failed, teardown of this fixture will dump all the DB and collect to the test servers
+    '''
+    if hasattr(request.node, 'rep_call') and request.node.rep_call.failed:
+        db_dump_home = "./logs"
+        db_dump_base = db_dump_home + "/db_dump"
+        db_dump_path = os.path.join(db_dump_base, request.module.__name__, request.node.name)
+        db_dump_tarfile = "{}.tar.gz".format(db_dump_base)
+
+        # Collect DB config
+        dbs = set()
+        result = duthosts[0].shell("cat /var/run/redis/sonic-db/database_config.json")
+        db_config = json.loads(result['stdout'])
+        for db in db_config['DATABASES']:
+            db_id = db_config['DATABASES'][db]['id']
+            dbs.add(db_id)
+
+        # Collect DB dump
+        duthosts.file(path=db_dump_path, state="directory")
+        for i in dbs:
+            duthosts.shell("redis-dump -d {} -y -o {}/{}".format(i, db_dump_path, i))
```

Comment:
On duthost, it would be better to generate the dump files to "/tmp".

Reply:
OK，I will change the path